test/integration/logic/resolvable/UnifyRelationConcludableTest.java

```diff
@@ -787,6 +856,314 @@ public void relation_more_players_than_rule_relation_fails_unify() {
+                String.format("{(employer: $x, employee: $y); $x iid %s;}", ptOrganisation),
+                String.format("{(employer: $x, employee: $y); $x iid %s;}", drivingHire),
+//3
+                String.format("{(employer: $x, employee: $y); $y iid %s;}", person),
+                String.format("{(employer: $x, employee: $y); $y iid %s;}", student),
+                String.format("{(employer: $x, employee: $y); $y iid %s;}", studentDriver),
+//6
+                String.format("{(employer: $x, employee: $y); $x iid %s;$y iid %s;}", person, student), //1-2
+                String.format("{(employer: $x, employee: $y); $x iid %s;$y iid %s;}", person, studentDriver), //1-
+                String.format("{(employer: $x, employee: $y); $x iid %s;$y iid %s;}", person, drivingHire),//1-3
+                //9
+                String.format("{(employer: $x, employee: $y); $x iid %s;$y iid %s;}", student, studentDriver), //2-31
+                String.format("{(employer: $x, employee: $y); $x iid %s;$y iid %s;}", student, ptOrganisation), //2-2
+                String.format("{(employer: $x, employee: $y); $x iid %s;$y iid %s;}", student, drivingHire), //2-3
+                //12
+                String.format("{(employer: $x, employee: $y); $x iid %s;$y iid %s;}", studentDriver, drivingHire), //3-3
+                String.format("{(employer: $x, employee: $y); $x iid %s;$y iid %s;}", studentDriver, studentDriver)
+        );
+
+        List<String> conclusions = Lists.newArrayList(
+                "(employer: $p, employee: $q) isa employment",
+                "(employer: $p, employee: $p) isa employment",
+                "(part-time-employer: $p, part-time-employee: $q) isa part-time-employment",
+                "(taxi: $p, day-shift-driver: $q) isa part-time-driving"
+        );
+        unifiable(conclusions.get(0), "{$p isa person; $q isa person;}", parents, Lists.newArrayList(3, 4, 5, 6, 7, 9, 13));
+        unifiable(conclusions.get(0), "{$p isa student; $q isa student-driver;}", parents, Lists.newArrayList(5, 9, 13));
+
+        //due to reflexivity all variants with different ids are not satisfiable
+        unifiable(conclusions.get(1), "{$p isa person;}", parents, Lists.newArrayList(3, 4, 5));
+        unifiable(conclusions.get(1), "{$p isa student;}", parents, Lists.newArrayList(4, 5 ));
+
+        unifiable(conclusions.get(2), "{$p isa part-time-organisation;$q isa student;}", parents, Lists.newArrayList(1, 2, 4, 5));
+        unifiable(conclusions.get(2), "{$p isa student;$q isa student-driver;}", parents, Lists.newArrayList(5, 9, 13));
+
+        unifiable(conclusions.get(3), "{$p isa driving-hire; $q isa student-driver;}", parents, Lists.newArrayList(2, 5, 12));
+        unifiable(conclusions.get(3), "{$p isa part-time-organisation; $q isa student;}", parents, Lists.newArrayList(1, 2, 4, 5, 10, 11, 12));
+    }
+
+    @Test
+    public void relationWithTypeSubstitution() {
```

Comment:
i'd say this test is actually `relations_with_exact_relation_instance`? the type is a by-product of the instance being known